### PR TITLE
feat: album player

### PR DESCRIPTION
**Pull request by Vibes-INS**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**The PR fulfills these requirements:**

- [ ] It's submitted to the `develop` branch(or to a previous version branch), _not_ the `master` branch
- [ ] All tests are passing
- [ ] Build success

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature

**Other information:**

### CODE REVIEW

File: src/views/AlbumPlayer/hooks/useAudioPlayer.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  )
+
+  const onPlay = useCallback(() => {
+    const el = audioRef.current
+    if (!el) return
+    el.play()
+  }, [])
+
+  useEffect(() => {
+    setSrc(list[index])
```
Comment (yuche) [inline src/views/AlbumPlayer/hooks/useAudioPlayer.tsx:45]: 你这个 src 是不是用 memo 生成更好

File: src/views/AlbumPlayer/index.tsx [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+    duration,
+    isPlaying,
+    onChangeProgress,
+  } = useAudioPlayer(list)
+
+  useEffect(() => {
+    if (isPlaying) {
+      setTimeout(() => {
+        setIsCdPlaying(true)
+      }, 200)
```
Comment (yuche) [inline src/views/AlbumPlayer/index.tsx:151]: 为什么是 200ms

File: src/views/AlbumPlayer/index.tsx [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+
+  const progress = isNaN(currentTime / duration) ? 0 : currentTime / duration
+  const cdProgress = Number(
+    ((1 / list.length) * index + (1 / list.length) * progress).toFixed(5)
+  )
+  const armRotate =
+    (ARM_RUN_RANGE[1] - ARM_RUN_RANGE[0]) * cdProgress + ARM_RUN_RANGE[0]
+
+  const isClosedCoverAnimation = useObservable(
+    () => timer(1500).pipe(map(() => true)),
```
Comment (yuche) [inline src/views/AlbumPlayer/index.tsx:165]: 1500ms 和后面的 3000ms 是什么意思，如果是动画效果时长，能不能用 onAnimationEnd 之类的回调做？

File: src/views/AlbumPlayer/index.tsx (hunk 2 above)
Comment (Vibes-INS) [reply src/views/AlbumPlayer/index.tsx:151]: 这里的200毫秒是为了等撞针放到唱片上面

Comment (yuche) [reply src/views/AlbumPlayer/index.tsx]: 那你把 200 做成一个常量吧，起个有意义的名字

**Review by yuche — DISMISSED** [review]

**Review by ybian19 — APPROVED** [review]

**Review by yuche — APPROVED** [review]

**Pull request merged**